shell/platform/common/cpp/client_wrapper/include/flutter/event_stream_handler.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+
+#include "event_sink.h"
+
+namespace flutter {
+
+// Handler of stream setup and tear-down requests.
+// Implementations must be prepared to accept sequences of alternating calls to
+// onListen() and onCancel(). Implementations should ideally consume no
+// resources when the last such call is not onListen(). In typical situations,
+// this means that the implementation should register itself with
+// platform-specific event sources onListen() and deregister again onCancel().
+template <typename T>
+struct StreamHandler {
+  // Handles a request to set up an event stream.
+  // |arguments| is stream configuration arguments and
+  // |events| is an EventSink for emitting events to the Flutter receiver.
+  using OnListen = std::function<void(const T* arguments,
```

Comment:
These need a way to communicate errors, as with the Java and ObjC versions—again, please don't deviate from the existing versions of these APIs with a strong justification.

Since we don't already have an error object, I think you'll need to make a simple `StreamHandlerError` object with the same arguments as an error envelope needs, and have these return a (possibly null) `std::unique_ptr<StreamHandlerError<T>>`. (It would be nice if we could have a common error object like ObjC, but this version will need the details to be a `std::unique_ptr<T>`, and the error calls take `const T*` for cases where it's not efficient to pass ownership, so that's not workable.)

Reply:
> These need a way to communicate errors, as with the Java and ObjC versions—again, please don't deviate from the existing versions of these APIs with a strong justification.

At first, I thought that Java (EventCHannel.java) and Objective-C (FlutterChannels.mm) had different specifications. But when I thought about it, I understood what you said. Thank you.